packages/tools/pluggable-widgets-tools/configs/rollup-plugin-collect-native-deps.js

```diff
@@ -0,0 +1,110 @@
+import { basename, dirname, extname, join } from "path";
+import { copy, readJson, writeJson } from "fs-extra";
+import { promises } from "fs";
+import { rm } from "shelljs";
+
+export function collectNativeDependencies({ outputDir, externals, widgetName }) {
+    let firstRun = true;
+    const nativeDependencies = [];
+    const nodeModulesPath = join(outputDir, "node_modules");
+    return {
+        name: "collect-native-deps",
+        async resolveId(source) {
+            if (source.startsWith(".")) {
+                return null;
+            }
+
+            try {
+                const packageFilePath = require.resolve(`${source}/package.json`);
+                const packageDir = dirname(packageFilePath);
+
+                if (await hasNativeCode(packageDir)) {
+                    if (!nativeDependencies.some(x => x.name === source)) {
+                        nativeDependencies.push({ name: source, dir: packageDir });
+                    }
+                    return { id: source, external: true };
+                }
+                return null;
+            } catch (e) {
+                return null;
```

Comment:
Should we throw an error here as `hasNativeCode` didn't run successfully?

Reply:
It is already inside the try so if its promise fail it will inform to rollup to just continue bundling instead of define as external and copy the lib